ui/webui/src/components/localization/InstallationLanguage.jsx

```diff
@@ -103,7 +103,8 @@ class LanguageSelector extends React.Component {
     }
 
     updateDefaultSelection () {
-        const languageId = this.state.lang.split("_")[0];
+        // First split removes the charset and the second removes optional region code.
+        const languageId = this.state.lang.split(".")[0].split("_")[0];
```

Comment:
I think you should be able to get the data with `getLocaleData` and retrieve the language id with `getLanguageId`.